### ISSUE TITLE
Consul Connect compatibility with envoy 1.10.0

### ISSUE THREAD

**Issue by danielehc** [hashicorp/consul-helm#370]
## Issue

While trying to deploy applications in a k8s cluster with Consul Connect deployed using the following values:

```yaml
global:
  domain: consul
  datacenter: dc1
  image: "consul:1.7.1"
server:
  replicas: 1
  bootstrapExpect: 1
  storage: 64Mi
  storageClass: local-path
client:
  enabled: true
  grpc: true
ui:
  enabled: true
connectInject:
  enabled: true
```

The pods are unable to start:

```
$ ./kubectl apply -f blueprints/learn-consul-service-mesh/k8s_config/api.yml 
deployment.apps/api-deployment-v1 created
```

```
$ kubectl get pods --all-namespaces 
NAMESPACE              NAME                                                              READY   STATUS             RESTARTS   AGE
kube-system            metrics-server-6d684c7b5-j6sfv                                    1/1     Running            0          4h42m
kube-system            local-path-provisioner-58fb86bdfd-q66m5                           1/1     Running            0          4h42m
kube-system            coredns-d798c9dd-sl5xk                                            1/1     Running            0          4h42m
kubernetes-dashboard   dashboard-metrics-scraper-566cddb686-ncg9p                        1/1     Running            0          4h41m
kubernetes-dashboard   kubernetes-dashboard-77f85889d5-psclt                             1/1     Running            0          4h41m
default                hashicorp-consul-connect-injector-webhook-deployment-5c48dbgl8t   1/1     Running            0          2m43s
default                hashicorp-consul-server-0                                         1/1     Running            0          2m42s
default                hashicorp-consul-xsl92                                            1/1     Running            0          2m44s
default                api-deployment-v1-85cc8c9977-9wkkp                                2/3     CrashLoopBackOff   1          68s
```

Checking the logs for the containers I see the following:

```
$ kubectl logs api-deployment-v1-85cc8c9977-9wkkp consul-connect-envoy-sidecar
[2020-03-04 16:37:17.484][000001][critical][main] [source/server/server.cc:86] error initializing configuration '/consul/connect-inject/envoy-bootstrap.yaml': Unable to parse JSON as proto (INVALID_ARGUMENT:layered_runtime: Cannot find field.): {"dynamic_resources":{"ads_config":{"grpc_services":{"envoy_grpc":{"cluster_name":"local_agent"},"initial_metadata":[{"value":"","key":"x-consul-token"}]},"api_type":"GRPC"},"cds_config":{"ads":{}},"lds_config":{"ads":{}}},"layered_runtime":{"layers":[{"static_layer":{"envoy.deprecated_features:envoy.config.filter.network.http_connection_manager.v2.HttpConnectionManager.Tracing.operation_name":true,"envoy.deprecated_features:envoy.config.trace.v2.ZipkinConfig.HTTP_JSON_V1":true,"envoy.deprecated_features:envoy.api.v2.Cluster.tls_context":true},"name":"static_layer"}]},"admin":{"address":{"socket_address":{"port_value":19000,"address":"127.0.0.1"}},"access_log_path":"/dev/null"},"static_resources":{"clusters":[{"hosts":[{"socket_address":{"port_value":8502,"address":"10.5.0.2"}}],"name":"local_agent","connect_timeout":"1s","type":"STATIC","http2_protocol_options":{}}]},"node":{"metadata":{"envoy_version":"1.13.0","namespace":"default"},"id":"api-deployment-v1-85cc8c9977-9wkkp-api-sidecar-proxy","cluster":"api"},"stats_config":{"use_all_default_tags":true,"stats_tags":[{"tag_name":"consul.custom_hash","regex":"^cluster\\.((?:([^.]+)~)?(?:[^.]+\\.)?[^.]+\\.[^.]+\\.[^.]+\\.[^.]+\\.[^.]+\\.consul\\.)"},{"tag_name":"consul.service_subset","regex":"^cluster\\.((?:[^.]+~)?(?:([^.]+)\\.)?[^.]+\\.[^.]+\\.[^.]+\\.[^.]+\\.[^.]+\\.consul\\.)"},{"tag_name":"consul.service","regex":"^cluster\\.((?:[^.]+~)?(?:[^.]+\\.)?([^.]+)\\.[^.]+\\.[^.]+\\.[^.]+\\.[^.]+\\.consul\\.)"},{"tag_name":"consul.namespace","regex":"^cluster\\.((?:[^.]+~)?(?:[^.]+\\.)?[^.]+\\.([^.]+)\\.[^.]+\\.[^.]+\\.[^.]+\\.consul\\.)"},{"tag_name":"consul.datacenter","regex":"^cluster\\.((?:[^.]+~)?(?:[^.]+\\.)?[^.]+\\.[^.]+\\.([^.]+)\\.[^.]+\\.[^.]+\\.consul\\.)"},{"tag_name":"consul.routing_type","regex":"^cluster\\.((?:[^.]+~)?(?:[^.]+\\.)?[^.]+\\.[^.]+\\.[^.]+\\.([^.]+)\\.[^.]+\\.consul\\.)"},{"tag_name":"consul.trust_domain","regex":"^cluster\\.((?:[^.]+~)?(?:[^.]+\\.)?[^.]+\\.[^.]+\\.[^.]+\\.[^.]+\\.([^.]+)\\.consul\\.)"},{"tag_name":"consul.target","regex":"^cluster\\.(((?:[^.]+~)?(?:[^.]+\\.)?[^.]+\\.[^.]+\\.[^.]+)\\.[^.]+\\.[^.]+\\.consul\\.)"},{"tag_name":"consul.full_target","regex":"^cluster\\.(((?:[^.]+~)?(?:[^.]+\\.)?[^.]+\\.[^.]+\\.[^.]+\\.[^.]+\\.[^.]+)\\.consul\\.)"},{"fixed_value":"api","tag_name":"local_cluster"}]}}
[2020-03-04 16:37:17.484][000001][info][main] [source/server/server.cc:507] exiting
```

## Workaround

I noticed that using a different version of Envoy the same works properly.

Here my working configuration file:

```yaml
global:
  domain: consul
  datacenter: hashidc1
  image: "consul:1.7.1"
server:
  replicas: 1
  bootstrapExpect: 1
  storage: 64Mi
  storageClass: local-path
client:
  enabled: true
  grpc: true
ui:
  enabled: true
connectInject:
  enabled: true
  imageEnvoy: envoyproxy/envoy:v1.13.1
```
I tested the workaround with 1.12.0, 1.12.2, 1.12.3, 1.13.0 too and seems to work.

**Comment by lkysow** [comment]
We will document this requirement in https://github.com/hashicorp/consul-helm/pull/365

**Comment by lkysow** [comment]
Closed by #365
